oauthproxy_test.go

```diff
@@ -837,3 +838,36 @@ func TestRequestSignaturePostRequest(t *testing.T) {
 	assert.Equal(t, 200, st.rw.Code)
 	assert.Equal(t, st.rw.Body.String(), "signatures match")
 }
+
+func TestGetRedirect(t *testing.T) {
+	options := NewOptions()
+	_ = options.Validate()
+	require.NotEmpty(t, options.ProxyPrefix)
+	proxy := NewOAuthProxy(options, func(s string) bool { return false })
+
+	tests := []struct {
+		name             string
+		url              string
+		expectedRedirect string
+	}{
+		{
+			name:             "request outside of ProxyPrefix redirects to original URL",
+			url:              "/foo/bar",
+			expectedRedirect: "/foo/bar",
+		},
+		{
+			name:             "request under of ProxyPrefix redirects to original URL",
```

Comment:
This comment isn't quite right
```suggestion
			name:             "request under ProxyPrefix redirects to root",
```